### PR TITLE
Adding CircularMapper/realignsamfile

### PR DESCRIPTION
**Pull request by jbv2**
Adds module for circularmapper realignsamfile.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by TCLamnidis** [comment]
Had to add entire BWA workflow in the tests >.< 
This SHOULD work, but nf-tests complains that the `.join` operation cannot happen as it only works on channels. Not sure why. 
I also tried applying a `map` to the `BWA_ALN.out.bam` channel, and add the fastq file in the middle of it, but that also complained.

File: modules/nf-core/circularmapper/realignsamfile/main.nf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    output:
+    tuple val(meta), path("*_realigned.bam")    , emit: bam
+    path "versions.yml"                         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}" // Module actually has a hard-coded output of {input}_realigned.bam
```
Comment (SPPearce) [inline modules/nf-core/circularmapper/realignsamfile/main.nf:24]: Can we mv the output file to include the prefix then.

File: modules/nf-core/circularmapper/realignsamfile/meta.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      description: Realigned BAM file
+      pattern: "*.bam"
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+authors:
+  - "@shyama-mama"
+  - "@jbv2"
+  - "TCLamnidis"
```
Comment (SPPearce) [inline modules/nf-core/circularmapper/realignsamfile/meta.yml:58]: This is missing a @

File: tests/config/pytest_modules.yml
Comment (SPPearce) [inline tests/config/pytest_modules.yml:1]: Don't want to readd the pytest for circulargenerator ;)

File: modules/nf-core/circularmapper/realignsamfile/main.nf (hunk 1 above)
Comment (TCLamnidis) [reply modules/nf-core/circularmapper/realignsamfile/main.nf]: Added!

**Review by jfy133 — COMMENTED** [review]
Looks like you still have pytest files still in there, you should remove the relevant files in 'tests/' and also the entry in the pytest_tags.yml (or whatever it is called, doing this from my phone)

Otherwise looks ok to me

File: modules/nf-core/circularmapper/realignsamfile/tests/nextflow.config
Comment (SPPearce) [inline modules/nf-core/circularmapper/realignsamfile/tests/nextflow.config:1]: Do you need this config file anymore?

Comment (TCLamnidis) [reply modules/nf-core/circularmapper/realignsamfile/tests/nextflow.config:1]: I guess strictly not, though it tested that the prefix was correctly picked up

**Review by SPPearce — APPROVED** [review]

**Comment by SPPearce** [comment]
Looks like the renaming to use the prefix needs to be conditional on the name of the file

**Comment by TCLamnidis** [comment]
indeed. good catch! Curious that it only got flagged on conda and not the other profiles 😅

**Comment by SPPearce** [comment]
> indeed. good catch! Curious that it only got flagged on conda and not the other profiles 😅

Yeah, I don't know why exactly. It doesn't throw an error on my macbook terminal, so presumably something about the conda enviroment is set differently.

**Review by SPPearce — APPROVED** [review]

**Pull request merged**